### PR TITLE
Add CMakeLists.txt to support PAL Tools

### PR DESCRIPTION
**Pull request by teetak01**

### CODE REVIEW

**Comment by teetak01** [comment]
Here is cleaned up version of the other PR. This should be enough.

**Comment by teetak01** [comment]
@tommikas @jupe

**Comment by teetak01** [comment]
Added yotta_ignore for backwards compatibility

**Comment by tommikas** [comment]
I don't have experience with cmake either, but this is sliiightly failing to build at the moment.

**Comment by teetak01** [comment]
Fixed the syntax in .yotta_ignore

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include/)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../nanostack-libservice/mbed-client-libservice/)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../nanostack-libservice/)
+
+
+FILE(GLOB MBED_TRACE_SRC
+	"${CMAKE_CURRENT_SOURCE_DIR}/source/*.c"
+
+	)
```
Comment (tommikas) [inline CMakeLists.txt]: Couldn't the above 4 lines be just one line?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+INCLUDE(CMakeForceCompiler)
+
+cmake_minimum_required (VERSION 2.8)
+SET(CMAKE_SYSTEM_NAME Generic)
+
+project(mbedTrace)
+
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include/)
```
Comment (tommikas) [inline CMakeLists.txt]: Is this correct? There's no `include` directory here, the header is in `mbed-trace`.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/)
```
Comment (tommikas) [inline CMakeLists.txt]: Does cmake need this? There's no source files directly in the root directory.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+cmake_minimum_required (VERSION 2.8)
+SET(CMAKE_SYSTEM_NAME Generic)
+
+project(mbedTrace)
+
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/mbed-trace/)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../nanostack-libservice/mbed-client-libservice/)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../nanostack-libservice/)
```
Comment (jupe) [inline CMakeLists.txt]: why two includes from libservice?

**Pull request merged**